src/BottomNavigation/BottomNavigation.react.js

```diff
@@ -44,20 +42,67 @@ function getStyles(props, context) {
 */
 
 class BottomNavigation extends PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = {
+            moveAnimated: new Animated.Value(0 - this.context.uiTheme.bottomNavigation.container.height),
+        };
+    }
+
+    componentDidMount() {
+        Animated.timing(this.state.moveAnimated, {
+            toValue: 0,
+            duration: 225,
+            easing: Easing.bezier(0.0, 0.0, 0.2, 1),
+        }).start();
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.shouldShow !== this.props.shouldShow && nextProps.shouldShow === true) {
+            this._show();
+        }
+
+        if (nextProps.shouldHide !== this.props.shouldHide && nextProps.shouldHide === true) {
```

Comment:
There should be just one property to control if the bottom navigation is visible or not ...

[Modal](https://facebook.github.io/react-native/docs/modal.html#visible) uses `visible`
[Status bar](https://facebook.github.io/react-native/docs/statusbar.html#hidden) uses `hidden`

So, pick one of them ;) Status bar is probably more similar to bottom navigation. And please, give it to `propTypes` :)